internal/pkg/template/templates/app/app.yml

```diff
@@ -188,7 +188,7 @@ Resources:
     Type: AWS::Route53::RecordSet
     Condition: DelegateDNS
     Properties:
-      HostedZoneName: !Sub ${AppDomainName}.
+      HostedZoneId: !Sub ${AppDomainHostedZoneID}
```

Comment:
I'm worried that since [this](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-route53-recordset.html#cfn-route53-recordset-hostedzoneid) is a field that'll trigger resource "replacement". Can you verify if it'll be a problem for current users when they do `app upgrade` (or any other command that'll trigger the app stack (not stackset instance)) to be updated?

Reply:
That's a good point. I will test this scenario.

Reply:
I tested this out for both scenarios i.e both upgrade and downgrade of copilot apps using `copilot app upgrade` 
This works well with out any potential issues.